### PR TITLE
HOTFIX: WindowedStreamPartitioner does not provide topic name to serializer

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
\cc @dguy @enothereska 
patch for `0.10.2` is #2776

**Comment by mjsax** [comment]
fix for failing tests are #2778 and  #2781

File: streams/src/main/java/org/apache/kafka/streams/state/StateSerdes.java [hunk 1]
```diff
@@ -73,6 +101,11 @@ public StateSerdes(String stateName,
         this.valueSerde = valueSerde;
     }
 
+    public StateSerdes<K, V> setApplicationId(final String applicationId) {
```
Comment (dguy) [inline streams/src/main/java/org/apache/kafka/streams/state/StateSerdes.java]: I'm not sure about this. It seems to be used only in the where we are using `WindowStoreUtils.INNER_SERDES`. I think i'd prefer to just create a new instance for each of those cases and do away with the `static final`

**Comment by enothereska** [comment]
LGTM. Would help reviewing if we separate actual change from other cleanup, e.g., two commits. Thanks.
1. fix
2. cleanup, including using `final` etc

**Comment by mjsax** [comment]
@eno I understand this -- but it's so much extra effort to split PRs... And if we don't cleanup in small pieces like this, we won't get there at all :( Or we need pure cleanup commits -- but I guess nobody is going to do this.

**Comment by mjsax** [comment]
Updated this.

**Comment by dguy** [comment]
@mjsax @enothereska i think doing small cleanups a long the way is the appropriate thing to do. Having separate PRs just for minor cleanups that can be done with another PR is a bit silly. Refactoring should be a constant effort. The trick is choosing to not go too far, which admittedly can be difficult at times.

**Comment by enothereska** [comment]
@mjsax @dguy I'm not suggesting separate PRs, just separate commits as part of the same PR. So in the PR we can see the commit history (fix, followed by cleanup, or vice versa). Right now for every 10 line fix there are another 100 lines of cleanup making it hard to see the exact fix. Changing 39 files for a fix like this makes reviewing hard and unnecessarily long. Again, not separate PRs, just a bit more care with commit order.

**Comment by dguy** [comment]
@enothereska oh - fair enough.

**Comment by enothereska** [comment]
LGTM btw

**Comment by mjsax** [comment]
Call for final review and merging \cc @gwenshap (we want this for `0.10.2.1` -- see https://github.com/apache/kafka/pull/2776 )

**Comment by mjsax** [comment]
Retest this please.

**Review by dguy — APPROVED** [review]
LGTM

**Review by ijuma — APPROVED** [review]
Thanks for the PR, LGTM.

**Pull request closed without merging**